### ISSUE TITLE
Vandalism by "Niwre Erv Apotseg"

### ISSUE THREAD

**Issue by TagaSanPedroAko** [OSMPH/papercut_fix#60]
Could someone watch and revert the edits of this this user, "[Niwre Erv Apotseg](https://www.openstreetmap.org/user/Niwre%20Erv%20Apotseg), which is behind the latest vandalism, possibly the worst and the most hideous we encountered after those by "bryanpiczon" and connected accounts (see #8), and Joey Samson (and suspected socks like "Lian Spurgeon Las Pinas", see #47 and #50 )?

Some notes:
-User name is a reverse reading of "Erwin Vre Gestopa".
-User was blocked for 5 days by the DWG.
-User joined OSM late February, and first edits are in Davao City. Since March 1, has been involved in large-scale disruption. "Timmy_Tesseract", @maning, @seav, and @ianlopez1115 have been monitoring the user since then.
-Edit comments are mostly nonsensical alphabet soup.
-User is mostly interested with creek names and fictional highways. Some of the user's edits added references to Cebuano/Bisaya profanity and the COVID-19 pandemic.

**Comment by TagaSanPedroAko** [comment]
Partially reverted some edits at this changesets:
-https://www.openstreetmap.org/changeset/82695840
-https://www.openstreetmap.org/changeset/82696272
-https://www.openstreetmap.org/changeset/82696725

**Comment by TagaSanPedroAko** [comment]
Tried to revert a bunch of edits by the user, but have accidentally deleted some road sections that do exist. I think they were split off an already mapped road by the user while adding the fictional features.

**Comment by ianlopez1115** [comment]
Cleaned up edits in the Taal Volcano area ([here](https://www.openstreetmap.org/changeset/82669801)) and Talim Island ([here](https://www.openstreetmap.org/changeset/82663815))

**Comment by SomeoneElseOSM** [comment]
Blocked again in 
https://www.openstreetmap.org/user_blocks/3587
(7 days)

Revert of everything not already reverted in
https://www.openstreetmap.org/changeset/82733493

**Comment by DearPaulPH** [comment]
@SomeoneElseOSM It seems that PH-630 (Philippine Highway Route 630) and most Bus/Jeepney routes (including DLTBCo, Superlines) got affected as well. Sadly, without the Mapillary imagery, verification efforts for the bridge name will be harder. Bus stop tweaks will be much easier via JOSM.

Note: bridge:name tags will be for the bridge name while the name tags will be for the common highway name

**Comment by TagaSanPedroAko** [comment]
@DearPaulPH Can you better specify the routes that got mangled here? The user's edits are really hard to fix.

**Comment by DearPaulPH** [comment]
@TagaSanPedroAko I'll manage the rest of the Tiwi-Tabaco-Legazpi portion of PH-630, including the whole route relation, and it will be fixed.

**Comment by DearPaulPH** [comment]
@TagaSanPedroAko All route relations (road, bus) for PH-630 got fixed already with the changeset 82740955, together with the JOSM Validator error fixes, addendum on boundary relations for Sagñay town in Camarines Sur (subject to OSMPH review), bus depot in Tabaco City. Here's the link: https://www.openstreetmap.org/changeset/82740955

**Comment by SomeoneElseOSM** [comment]
Just for info, in case anyone isn't already doing this:

If you want to visualise a way or relation at a certain point in time you can do so via mapki and Overpass, so for example for https://www.openstreetmap.org/relation/10650252 : 

http://osm.mapki.com/history/relation.php?id=10650252

(pick a date from there and use it in e.g.)

https://overpass-turbo.eu/s/S0I

**Comment by TagaSanPedroAko** [comment]
Done. 🚀

Closing this now, but can reopen this once user resurfaces again.

**Comment by TagaSanPedroAko** [comment]
That guys back again. Last edit is 14 days ago. Will try to revert.

**Comment by Timmy-Tesseract** [comment]
And he's back again. 
https://www.openstreetmap.org/user/ErvinJameSPogiBiancaPanget

How do you guys usually fix vandalism like this? Is it possible to automatically revert all of his recent changesets before anything is touched by others? Or is there some kind of tool that helps to review all individual ways that were affected?

I'd be happy to help cleaning up.

**Comment by SomeoneElseOSM** [comment]
> Is it possible to automatically revert all of his recent changesets before anything is touched by others? 

If you can detect it - sure.  I've done that sort of thing with "expected" prank changes to known objects.  "Random" edits (or even just prank edits to random objects) are more difficult.

> Or is there some kind of tool that helps to review all individual ways that were affected?

If you want to review everything as you go, then JOSM lets you do that - but JOSM's reverter can get confused in very complicated cases.

Please drop the DWG an email at data@osmfoundation.org with details of any user that you believe to be a sock-puppet of a previously problematical user, explaining what it is that they are doing that is a problem.

: Andy (from the DWG)

**Comment by Timmy-Tesseract** [comment]
> Please drop the DWG an email at [data@osmfoundation.org](mailto:data@osmfoundation.org) with details of any user that you believe to be a sock-puppet of a previously problematical user, explaining what it is that they are doing that is a problem.

Thank you Andy, done that. 

Considering that it's very obvious that the user is a sock-puppet, is it okay if I will start reverting the changesets while referring to https://www.openstreetmap.org/user_blocks/3587? 

Or should I wait until this account is blocked too?

**Comment by SomeoneElseOSM** [comment]
On 13/02/2021 08:46, Timmy-Tesseract wrote:
> Considering that it's very obvious that the user is a sock-puppet, is it okay if I will start reverting the changesets while referring to https://www.openstreetmap.org/user_blocks/3587?

It looks like you've reverted those already this morning.  Thanks for doing that.  Presumably it all reverted cleanly and there's nothing left to revert?

It appears that there's been no attempt to contact the mapper at all: http://resultmaps.neis-one.org/osm-discussion-comments?uid=12607985 .

You might say "well, _obviously_ it's the same person as the previous vandal, because they've done exactly the same sort of damage" (and you'd be very likely correct), but a changeset discussion comment saying "The bridge at way X is fictional; I know that by looking at XYZ imagery / local knowledge" would be really helpful to other local mappers who don't follow this github repository (maybe they just read the mailing list https://lists.openstreetmap.org/listinfo/talk-ph ).

Also, "not replying to other mappers" was one of the points mentioned in a previous block https://www.openstreetmap.org/user_blocks/3827 - if someone later says "but my mapping isn't fictional!" two of the three reasons for the original block (in this case) still apply.

One more thing - when fictional objects have been added it's really helpful to say an example of an object that is fictional and why it obviously fictional, to add to saying e.g. "https://www.openstreetmap.org/changeset/98804629 non-existent road construction added".

If you want to link to something that you're just about to revert, you can use a date query in overpass like https://overpass-turbo.eu/s/13Az (for just one way) and https://overpass-turbo.eu/s/13AA (for all ways in an area) or https://overpass-turbo.eu/s/13AB (for everything added or modified by a user at a date in an area.

Best Regards,

Andy

(from the DWG)

**Comment by Timmy-Tesseract** [comment]
> It looks like you've reverted those already this morning. Thanks for doing that. Presumably it all reverted cleanly and there's nothing left to revert?

I just partially reverted the changes of one single changeset. Other reverts have been done by apple data team. But they're exclusively reverting upgrades to road classifications, all of these segments will still need to be checked to remove the fictional names and ref values.

> It appears that there's been no attempt to contact the mapper at all: http://resultmaps.neis-one.org/osm-discussion-comments?uid=12607985 .

That's right. I just wasn't sure if this is still needed again, but it definitely makes sense as a reference for other editors and also to see if the vandal will communicate this time. I now commented on a few of the new changesets and also sent him a private message. 

Thanks for the overpass links. I'm still new to that tool, this will be very useful. I also did some experimenting with the JOSM reverter earlier so I think I can manage to do the cleanup myself.

**Comment by Timmy-Tesseract** [comment]
Quick update, new account is blocked too:
https://www.openstreetmap.org/user_blocks/4752

Big thanks again to Andy for doing most of the reverts. 

Will do the last remaining clean-ups this weekend, issue can be considered solved for now.

**Comment by Timmy-Tesseract** [comment]
It seems like he's already back with another sock-puppet account. Only three changesets so far, but suspiciously similar editing patterns and area of activity.

Attempt to contact mapper has been done, will wait for a few days before starting to revert.

[sveeb](http://hdyc.neis-one.org/?sveeb)

Edit: And another one [frur](http://hdyc.neis-one.org/?frur)

**Comment by Timmy-Tesseract** [comment]
Quick update on this... He changed the name of his latest account to [Frogs Are Ugly Rocks](http://hdyc.neis-one.org/?Frogs%20Are%20Ugly%20Rocks) and he keeps on messing up road classification and adding fictional/non-verifiable names. Like before he's mostly active in Mindanao, but some of his recent changesets have also affected NCR as well as an island belongig to Australia. Report at DWG is still pending.

**Comment by SomeoneElseOSM** [comment]
>  Report at DWG is still pending.

I can't see anything from the DWG end, suggesting that either someone's already dealing with it, someone ignored it or you reported via the website and it hasn't reached us yet.  if you've received a "this is an automated reply" reply then I'd suggest replying to that so that whoever's dealing with it knows that the ticket is still an issue.  If you reported via the website try emailing data@openstreetmap.org.

**Comment by Timmy-Tesseract** [comment]
> I can't see anything from the DWG end, suggesting that either someone's already dealing with it, someone ignored it or you reported via the website and it hasn't reached us yet. if you've received a "this is an automated reply" reply then I'd suggest replying to that so that whoever's dealing with it knows that the ticket is still an issue. If you reported via the website try emailing [data@openstreetmap.org](mailto:data@openstreetmap.org).

Thanks for following up on this. I've sent e-mails to data@osmfoundation.org, but didn't receive any confirmation if they were received. Will resend to the correct address.

**Comment by SomeoneElseOSM** [comment]
data@osmfoundation.org is supposed to work as well, but has a more aggressive spam filter on it.  You should have had a "this is an automated reply" reply now from the new ticket.

**Comment by DP24PH** [comment]
We might have to elevate it into an IP-based block in addition to the user block since it is considered a sock-puppet-related abuse. I'm now monitoring R4A as well into heightened alert to watch for unusual edits.

**Comment by TagaSanPedroAko** [comment]
I'm trying to focus on mapping Canada, and I have little to do with this issue, but I would agree this case clearly involves sock puppetry (same as with the Joey Samson case). I don't OSM do IP blocks as Wikipedia does.

**Comment by SomeoneElseOSM** [comment]
IP blocks are rarely useful, to be honest.  We can ask for them to be set up but what usually happens is that the offending address is in the dynamic pool of a major ISP or other large institution which obviously can't be blocked without major knock-on effects for other users.

**Comment by TagaSanPedroAko** [comment]
I'v just reverted edits around Sorsogon using Go Map!, but have placed the wrong comment (from an earlier edit in Batangas using the same app) due to the annoying X button on the upload screen that do often get pressed accidentally when correcting something on the comments.

@Timmy-Tesseract Any updates from DWG?

**Comment by Timmy-Tesseract** [comment]
> 
> @Timmy-Tesseract Any updates from DWG?

DWG hast done a complete revert and I cleaned up the remaining conflicts a few weeks ago. Issue can be closed for now and hopefully for good.

**Comment by SomeoneElseOSM** [comment]
> Issue can be closed for now and hopefully for good.

I wish I shared your optimism...

**Comment by Timmy-Tesseract** [comment]
> 
> I wish I shared your optimism...

It's more wishful thinking than optimism. 

Thanks again for your help!
